### DISCUSSION TITLE
sending ether value on contract write to payable function

### DISCUSSION THREAD

**Discussion by pmespresso** [wevm/wagmi#1052] [Ideas]
Hi guys, I've been really enjoying using the wagmi hooks but there is one feature request I'd like to add (even implement with some guidance).

The wagmi way to call a writeable function is with `usePrepareContractWrite` and then `useContractWrite`, something like: 
```
  const { config, error } = usePrepareContractWrite({
    addressOrName: '0xecb504d39723b0be0e3a9aa33d646642d1051ee1',
    contractInterface: wagmigotchiABI,
    functionName: 'feed',
  })
```
but the options don't include a `value` key to send Ether along for when calling a `payable` function.

e.g.
```
  const { config, error } = usePrepareContractWrite({
    addressOrName: '0xecb504d39723b0be0e3a9aa33d646642d1051ee1',
    contractInterface: wagmigotchiABI,
    functionName: 'feed',
    value: ethers.parseUnits(0.01)
  })
```

I'm wondering why, as from a DevEx perspective that's reasonable to expect.

If there is another way to do this, then the docs (or maybe the search) need to be updated because I haven't found anything other than `usePrepareSendTransaction` which is not what I need.

Again, happy to help implement if everyone is too busy. Thank you for reading my issue.

**Comment by pmespresso** [comment]
nevermind I found it in the overrides and I'm terribly embarrased 39 lashings tonight ggwp

**Reply by f1lander** [reply]
@pmespresso where is this docs to override?
